dev/com.ibm.ws.jaxrs.2.1.common/src/com/ibm/ws/jaxrs20/providers/jsonp/JsonPProvider.java

```diff
@@ -33,177 +36,89 @@
 import com.ibm.websphere.ras.Tr;
 import com.ibm.websphere.ras.TraceComponent;
 import com.ibm.ws.ffdc.annotation.FFDCIgnore;
-import com.ibm.ws.jaxrs20.component.LibertyJaxRsThreadPoolAdapter;
-import com.ibm.ws.jaxrs20.utils.ReflectUtil;
-import com.ibm.wsspi.classloading.ClassLoadingService;
 
-@SuppressWarnings("rawtypes")
 @Produces({ "application/json", "application/*+json" })
 @Consumes({ "application/json", "application/*+json" })
 @Provider
-public class JsonPProvider implements MessageBodyReader, MessageBodyWriter {
+public class JsonPProvider implements MessageBodyReader<Object>, MessageBodyWriter<Object> {
     private final static TraceComponent tc = Tr.register(JsonPProvider.class);
 
-    private final static String[] jsonpClasses = new String[] { "javax.json.Json", "javax.json.JsonArray", "javax.json.JsonException", "javax.json.JsonObject",
-                                                                "javax.json.JsonReader",
-                                                                "javax.json.JsonStructure", "javax.json.JsonWriter" };
-
-    private final static Map<String, Class<?>> jsonpClsMaps = new HashMap<String, Class<?>>();
-    private final static Map<String, Method> jsonpMethodMaps = new HashMap<String, Method>();
-
-    static {
-
-        ClassLoadingService cls = null;
-        ClassLoader cl = null;
-        try {
-            cls = LibertyJaxRsThreadPoolAdapter.getClassLoadingServiceref().getServiceWithException();
-            cl = cls.createThreadContextClassLoader(JsonPProvider.class.getClassLoader());
-        } catch (Throwable t) {
-            cl = JsonPProvider.class.getClassLoader();
-        } finally {
+    JsonProvider jsonProvider = null;
 
-            for (String clsName : jsonpClasses) {
-                Class<?> c = ReflectUtil.loadClass(cl, clsName);
-                if (c != null) {
-                    jsonpClsMaps.put(clsName, c);
-                }
-            }
-            if (cls != null) {
-                cls.destroyThreadContextClassLoader(cl);
-            }
-        }
+    public JsonPProvider(JsonProvider jsonProvider) {
+        this.jsonProvider = jsonProvider;
     }
 
     @Override
-    public long getSize(Object arg0, Class arg1, Type arg2, Annotation[] arg3, MediaType arg4) {
-
+    public long getSize(Object obj, Class<?> type, Type genericType, Annotation[] annotations, MediaType mediaType) {
```

Comment:
The `getSize` method is now a default interface method, so in JAX-RS 2.1, we don't need to implement/override it here.  I'm fine delivering this as-is, but let's remove this method when we make a further change to this class.